File: back-end/src/main/java/com/huyphan/dtos/NewPostDto.java
package com.huyphan.dtos;

import com.huyphan.models.enums.PostContentType;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class NewPostDto {

    private String title;

    private String mediaUrl;

    private String mediaType;

    private SectionDto section;

    private String tags;

    private String text;

    private PostContentType contentType;

    private boolean anonymous;

    private boolean notificationEnabled;
}

File: client/src/services/dtos/new-post-dto.ts
import { PostContentType } from '../../models/enums/post-content-type';

import SectionDto from './section-dto';

export default interface NewPostDto {
  readonly title?: string;

  readonly mediaUrl: string | null;

  readonly mediaType: string | null;

  readonly section: SectionDto;

  readonly tags: string;

  readonly text: string | null;

  readonly contentType: PostContentType;

  readonly anonymous: boolean;

  readonly notificationEnabled: boolean;

  readonly nsfw: boolean;
}
